src/input.rs

```diff
@@ -136,49 +136,23 @@ impl ButtonCode {
             KEY_POWER => ButtonCode::Power,
             KEY_HOME => ButtonCode::Home,
             KEY_LIGHT => ButtonCode::Light,
-            KEY_BACKWARD => resolve_button_direction(ButtonDirection::Backward, rotation, button_scheme),
-            KEY_FORWARD => resolve_button_direction(ButtonDirection::Forward, rotation, button_scheme),
+            KEY_BACKWARD => resolve_button_direction(LinearDir::Backward, rotation, button_scheme),
+            KEY_FORWARD => resolve_button_direction(LinearDir::Forward, rotation, button_scheme),
             _ => ButtonCode::Raw(code)
         }
     }
 }
 
-#[derive(Debug, Copy, Clone, Eq, PartialEq)]
-enum ButtonDirection {
-    Backward,
-    Forward
-}
-
-impl ButtonDirection {
-    fn flip(self) -> ButtonDirection {
-        match self {
-            ButtonDirection::Backward => ButtonDirection::Forward,
-            ButtonDirection::Forward => ButtonDirection::Backward,
-        }
+fn resolve_button_direction(mut direction: LinearDir, rotation: i8, button_scheme: ButtonScheme) -> ButtonCode {
+    if (rotation >= 2) ^ (button_scheme == ButtonScheme::Inverted) {
```

Comment:
`rotation >= 2` needs to be generalised as `CURRENT_DEVICE.should_invert_buttons(rotation)` where the relevant method is:
```rust
pub fn should_invert_buttons(&self, rotation: i8) -> bool {
	let sr = self.startup_rotation();
	let (_, dir) = self.mirroring_scheme();
	rotation == (4 + sr - dir) % 4 || rotation == (4 + sr - 2 * dir) % 4
}
```

And `button_scheme == ButtonScheme::Inverted` needs to be `CURRENT_DEVICE.orientation(rotation) == Orientation::Portrait && button_scheme == ButtonScheme::Inverted`: I don't see how it would make sense to apply the setting in landscape mode.

Reply:
> `rotation >= 2` needs to be generalised as `CURRENT_DEVICE.should_invert_buttons(rotation)` where the relevant method is:

Makes sense. Updated.

Reply:
> And `button_scheme == ButtonScheme::Inverted` needs to be `CURRENT_DEVICE.orientation(rotation) == Orientation::Portrait && button_scheme == ButtonScheme::Inverted`: I don't see how it would make sense to apply the setting in landscape mode.

As I mentioned in the initial PR message, I think this is a sensitive area so I'm taking the approach to be consistent with the inversion. Kobo themselves have updated their landscape inversion behavior in Nickel as of the *I think* December update. The inversion behavior in this PR works exactly as it does in Nickel. 

Personally I do not use landscape mode so I do not know what the consensus is. I made the `Button Scheme` setting a list for the intent that if people do wish for more granular inversion behavior, they can add it down the line (Inverted only for Portrait, Inverted only for Landscape, Inverted only for Portrait but not mirrored, etc).

Reply:
> I made the `Button Scheme` setting a list for the intent that if people do wish for more granular inversion behavior, they can add it down the line (Inverted only for Portrait, Inverted only for Landscape, Inverted only for Portrait but not mirrored, etc).

You haven't changed the `button_scheme` field of the `Settings` struct, so I'm not sure how it could be a *list*.

Reply:
Sorry, I meant the display of the settings here   https://github.com/baskerville/plato/blob/954f1d961547afe620cea57c31e9fc2d1f4698c5/src/view/common.rs#L113 where I've opted to not use a toggle similar to the display invert due to the potential of there being additional inversion schemes in the future.

Reply:
I think there's still a discrepancy here: if you want the user to be able to set the button scheme for each rotation value, then `button_scheme` needs to be an array: `[ButtonScheme; 4]`.

Reply:
I understand what you mean. I'd prefer to kick this decision down the road and keep the code in this open-flexible state where we haven't committed to either:
- handling button scheme inversion as a toggle
- adding the inversion option per rotation value

**until user desire manifests** since I do not know what the preference is. 

Implementing the inversion option per rotation value opens up some other design questions/discussions which I don't know are worth handling until user sentiment is known.

These questions would involve the best way to present these granular settings to the user:
- A straight list? `[Natural, Inverted, Landscape Inverted Only, Portrait Inverted Only]`
- A list with a toggle **per rotation value**?
- Displaying a toggle that only sets the inversion for the current orientation and rotation?

Obviously this is your product and if you wish me to just implement one of them, I can go ahead and do that. I just wanted to keep this decision open until more user feedback was gathered/known.

Reply:
My bad: I misread you. I'm also in favour of waiting for users to request more flexibility.